src/enlyze/client.py

```diff
@@ -139,56 +156,107 @@ def get_timeseries(
 
         :raises: |generic-error|
 
-        :returns: Timeseries data or ``None`` if API returned no data for the request
+        :returns: Timeseries data or ``None`` if the API returned
+            no data for the request
 
         """
 
-        if start.utcoffset() is None or end.utcoffset() is None:
-            logging.warning(
-                "Passing naive datetime is discouraged, assuming local timezone."
-            )
+        start, end, appliance_uuid = validate_timeseries_arguments(
+            start, end, variables
+        )
+
+        pages = self._client.get_paginated(
+            "timeseries",
+            api_models.TimeseriesData,
+            params={
+                "appliance": appliance_uuid,
+                "start_datetime": start.isoformat(),
+                "end_datetime": end.isoformat(),
+                "variables": ",".join(str(v.uuid) for v in variables),
+            },
+        )
+
+        timeseries_data = _get_timeseries_data_from_pages(pages)
+        if timeseries_data is None:
+            return None
+
+        return timeseries_data.to_user_model(
+            start=start,
+            end=end,
+            variables=variables,
+        )
+
+    def get_timeseries_with_resampling(
+        self,
+        start: datetime,
+        end: datetime,
+        variables: list[user_models.VariableWithResamplingMethod],
```

Comment:
> As we can see throughout, it complicates the handling of a "Variable" because we need to deal with both types now (mainly in the type annotations so far).

I disagree that it complicates anything, especially not the type annotations. I've defined a type alias for the annotations so should make for more DRY type declarations.

> Moreover, we won't need to educate the user how to use convert_to_variable_with_resampling_method() in order to generate a VariableWithResamplingMethod, so IMHO that would be a simpler to use interface.

Well, we can do the same like in your suggestion and omit it, they're still gonna have to convert a list of variables that they have to a dict of variables and resampling methods. This is just a convenience function, in both cases it will need to happen.

I don't think your suggestion provides any improvement over the status quo.

Reply:
We can remove the conversion function, in both cases, the user will have to do manipulation to go from variable to variable with a resampling method. The mechanism is literally the same, there's no way around it.